sky/data/data_utils.py

```diff
@@ -49,6 +51,24 @@ def split_gcs_path(gcs_path: str) -> Tuple[str, str]:
     return bucket, key
 
 
+def split_az_path(az_path: str) -> Tuple[str, str, str]:
+    """Splits Path into Storage account and Container names and Relative Path
+
+    Args:
+        az_path: str; Container Path, e.g. az://storage_account/container/
+
+    Returns:
+        str; Name of the container
+        str; paths of the file/directory defined within the container
+        str; Name of the storage account
+    """
+    path_parts = az_path.replace('az://', '').split('/')
+    storage_account_name = path_parts.pop(0)
+    container_name = path_parts.pop(0)
```

Comment:
After manually overriding the region of the storage account, I was able to get the spot controller to launch. However, launching the task on it failed with these logs:
```
(base) ➜  ~  sky spot logs --controller 1
D 05-01 13:34:46 skypilot_config.py:144] Using config path: /Users/romilb/.sky/config.yaml
D 05-01 13:34:46 skypilot_config.py:148] Config loaded:
D 05-01 13:34:46 skypilot_config.py:148] {'serve': {'controller': {'resources': {'cloud': 'kubernetes'}}},
D 05-01 13:34:46 skypilot_config.py:148]  'spot': {'controller': {'resources': {'cloud': 'azure'}}}}
D 05-01 13:34:46 skypilot_config.py:158] Config syntax check passed.
D 05-01 13:34:48 azure.py:645] az vm list --query "[?tags.\"ray-cluster-name\"=='sky-spot-controller-2ea485ea-2ea4'].id" -o json returned 0.
D 05-01 13:34:48 azure.py:645] **** STDOUT ****
D 05-01 13:34:48 azure.py:645] [
D 05-01 13:34:48 azure.py:645]   "/subscriptions/aa86df77-e703-453e-b2f4-955c3b33e534/resourceGroups/SKY-SPOT-CONTROLLER-2EA485EA-2EA4-EASTUS/providers/Microsoft.Compute/virtualMachines/ray-sky-spot-controller-2ea485ea-2ea4-head-6df2-c2120"
D 05-01 13:34:48 azure.py:645] ]
D 05-01 13:34:48 azure.py:645]
D 05-01 13:34:48 azure.py:645] **** STDERR ****
D 05-01 13:34:48 azure.py:645]
D 05-01 13:34:49 azure.py:665] az vm show -d --ids /subscriptions/aa86df77-e703-453e-b2f4-955c3b33e534/resourceGroups/SKY-SPOT-CONTROLLER-2EA485EA-2EA4-EASTUS/providers/Microsoft.Compute/virtualMachines/ray-sky-spot-controller-2ea485ea-2ea4-head-6df2-c2120 --query "powerState" -o json returned 0.
D 05-01 13:34:49 azure.py:665] **** STDOUT ****
D 05-01 13:34:49 azure.py:665] "VM running"
D 05-01 13:34:49 azure.py:665]
D 05-01 13:34:49 azure.py:665] **** STDERR ****
D 05-01 13:34:49 azure.py:665]
Tailing logs of job 1 on cluster 'sky-spot-controller-2ea485ea'...
I 05-01 20:34:53 log_lib.py:407] Start streaming logs for job 1.
INFO: Tip: use Ctrl-C to exit log streaming (task will not be killed).
INFO: Waiting for task resources on 1 node. This will block if the cluster is full.
INFO: All task resources reserved.
INFO: Reserved IPs: ['10.247.0.4']
(sky-2a8c-romilb, pid=8641) D 05-01 20:30:08 skypilot_config.py:144] Using config path: /home/azureuser/.sky/spot_tasks/sky-2a8c-romilb-779f.config_yaml
(sky-2a8c-romilb, pid=8641) D 05-01 20:30:08 skypilot_config.py:148] Config loaded:
(sky-2a8c-romilb, pid=8641) D 05-01 20:30:08 skypilot_config.py:148] {'serve': {'controller': {'resources': {'cloud': 'kubernetes'}}},
(sky-2a8c-romilb, pid=8641) D 05-01 20:30:08 skypilot_config.py:148]  'spot': {'controller': {'resources': {'cloud': 'azure'}}}}
(sky-2a8c-romilb, pid=8641) D 05-01 20:30:08 skypilot_config.py:158] Config syntax check passed.
(sky-2a8c-romilb, pid=8641) D 05-01 20:30:09 skypilot_config.py:144] Using config path: /home/azureuser/.sky/spot_tasks/sky-2a8c-romilb-779f.config_yaml
(sky-2a8c-romilb, pid=8641) D 05-01 20:30:09 skypilot_config.py:148] Config loaded:
(sky-2a8c-romilb, pid=8641) D 05-01 20:30:09 skypilot_config.py:148] {'serve': {'controller': {'resources': {'cloud': 'kubernetes'}}},
(sky-2a8c-romilb, pid=8641) D 05-01 20:30:09 skypilot_config.py:148]  'spot': {'controller': {'resources': {'cloud': 'azure'}}}}
(sky-2a8c-romilb, pid=8641) D 05-01 20:30:09 skypilot_config.py:158] Config syntax check passed.
(sky-2a8c-romilb, pid=8641) Process Process-1:
(sky-2a8c-romilb, pid=8641) Traceback (most recent call last):
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/multiprocessing/process.py", line 314, in _bootstrap
(sky-2a8c-romilb, pid=8641)     self.run()
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/multiprocessing/process.py", line 108, in run
(sky-2a8c-romilb, pid=8641)     self._target(*self._args, **self._kwargs)
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/site-packages/sky/spot/controller.py", line 387, in _run_controller
(sky-2a8c-romilb, pid=8641)     spot_controller = SpotController(job_id, dag_yaml, retry_until_up)
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/site-packages/sky/spot/controller.py", line 52, in __init__
(sky-2a8c-romilb, pid=8641)     self._dag, self._dag_name = _get_dag_and_name(dag_yaml)
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/site-packages/sky/spot/controller.py", line 40, in _get_dag_and_name
(sky-2a8c-romilb, pid=8641)     dag = dag_utils.load_chain_dag_from_yaml(dag_yaml)
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/site-packages/sky/utils/dag_utils.py", line 104, in load_chain_dag_from_yaml
(sky-2a8c-romilb, pid=8641)     task = task_lib.Task.from_yaml_config(task_config, env_overrides)
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/site-packages/sky/task.py", line 419, in from_yaml_config
(sky-2a8c-romilb, pid=8641)     storage_obj = storage_lib.Storage.from_yaml_config(storage[1])
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/site-packages/sky/data/storage.py", line 1004, in from_yaml_config
(sky-2a8c-romilb, pid=8641)     storage_obj = cls(name=name,
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/site-packages/sky/data/storage.py", line 492, in __init__
(sky-2a8c-romilb, pid=8641)     self._validate_storage_spec(name)
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/site-packages/sky/data/storage.py", line 715, in _validate_storage_spec
(sky-2a8c-romilb, pid=8641)     name = data_utils.split_az_path(source)[0]
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/site-packages/sky/data/data_utils.py", line 67, in split_az_path
(sky-2a8c-romilb, pid=8641)     container_name = path_parts.pop(0)
(sky-2a8c-romilb, pid=8641) IndexError: pop from empty list
(sky-2a8c-romilb, pid=8641) I 05-01 20:30:10 controller.py:473] Killing controller process 9375.
(sky-2a8c-romilb, pid=8641) I 05-01 20:30:10 controller.py:481] Controller process 9375 killed.
(sky-2a8c-romilb, pid=8641) I 05-01 20:30:10 controller.py:483] Cleaning up any spot cluster for job 1.
(sky-2a8c-romilb, pid=8641) Traceback (most recent call last):
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/runpy.py", line 196, in _run_module_as_main
(sky-2a8c-romilb, pid=8641)     return _run_code(code, main_globals, None,
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/runpy.py", line 86, in _run_code
(sky-2a8c-romilb, pid=8641)     exec(code, run_globals)
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/site-packages/sky/spot/controller.py", line 532, in <module>
(sky-2a8c-romilb, pid=8641)     start(args.job_id, args.dag_yaml, args.retry_until_up)
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/site-packages/sky/spot/controller.py", line 491, in start
(sky-2a8c-romilb, pid=8641)     _cleanup(job_id, dag_yaml=dag_yaml)
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/site-packages/sky/spot/controller.py", line 431, in _cleanup
(sky-2a8c-romilb, pid=8641)     dag, _ = _get_dag_and_name(dag_yaml)
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/site-packages/sky/spot/controller.py", line 40, in _get_dag_and_name
(sky-2a8c-romilb, pid=8641)     dag = dag_utils.load_chain_dag_from_yaml(dag_yaml)
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/site-packages/sky/utils/dag_utils.py", line 104, in load_chain_dag_from_yaml
(sky-2a8c-romilb, pid=8641)     task = task_lib.Task.from_yaml_config(task_config, env_overrides)
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/site-packages/sky/task.py", line 419, in from_yaml_config
(sky-2a8c-romilb, pid=8641)     storage_obj = storage_lib.Storage.from_yaml_config(storage[1])
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/site-packages/sky/data/storage.py", line 1004, in from_yaml_config
(sky-2a8c-romilb, pid=8641)     storage_obj = cls(name=name,
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/site-packages/sky/data/storage.py", line 492, in __init__
(sky-2a8c-romilb, pid=8641)     self._validate_storage_spec(name)
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/site-packages/sky/data/storage.py", line 715, in _validate_storage_spec
(sky-2a8c-romilb, pid=8641)     name = data_utils.split_az_path(source)[0]
(sky-2a8c-romilb, pid=8641)   File "/home/azureuser/miniconda3/lib/python3.10/site-packages/sky/data/data_utils.py", line 67, in split_az_path
(sky-2a8c-romilb, pid=8641)     container_name = path_parts.pop(0)
(sky-2a8c-romilb, pid=8641) IndexError: pop from empty list
```

Reply:
This is resolved with eaf3058